tests/rust/safe_abstraction/deque.rs

```diff
@@ -230,8 +230,8 @@ impl Deque {
     }
 
     pub fn pop_front<'a>(&'a mut self) -> i32
-        //@ req thread_token(?_t) &*& [?_q]lifetime_token(?a) &*& full_borrow(a, Deque_full_borrow_content(_t, self));
-        //@ ens thread_token(_t) &*& [_q]lifetime_token(a);
+        //@ req thread_token(?_t, ?m) &*& [?_q]lifetime_token(?a) &*& full_borrow(a, Deque_full_borrow_content(_t, self));
```

Comment:
I think it is more instructive for the reader to say TopMinus([]) here? But this makes me wonder if we should not introduce a separate predicate partial_thread_token(t, m) and define thread_token(t) = partial_thread_token(t, MaskTopMinus([])) so that people who don't know or care about nonatomic borrows are not bothered with these masks?

Reply:
I chose the second solution. safe_abstraction/deque.rs and tree.rs are back to using `thread_token(t)`